### PR TITLE
Tidy up whitespace, namespaces and dead code in Filters.h and Filters.cpp

### PR DESCRIPTION
**Pull request by tristanpenman**
This is a general cleanup PR for Filters.h and Filters.cpp. Whitespace is now much more consistent with the rest of the DSSTNE codebase, namespaces are fully qualified in the header file, and I was able to remove some code that was no longer necessary thanks to @sgkim126's very nice unique_ptr changes.

There is scope for further improvements, specifically removing 'hack' code, but that can be done later.

### CODE REVIEW

File: src/amazon/dsstne/utils/Filters.cpp [hunk 1]
```diff
@@ -17,94 +17,82 @@
...
     }
 }
 
-void AbstractFilter::updateRecords(float *xArray,unordered_map<int,float> *xFilter, int offSet, int width)
+/**
+ * @param xArray values to be filtered
+ * @param offset the starting global index of current xArray
+ * @param width the length of xArray
+ */
+void AbstractFilter::updateRecords(float *xArray, const unordered_map<int, float> *xFilter, int offset, int width)
```
Comment (rybakov) [inline src/amazon/dsstne/utils/Filters.cpp]: const int offset, const int width

File: src/amazon/dsstne/utils/Filters.cpp [hunk 2]
```diff
@@ -190,62 +176,49 @@ void SamplesFilter::loadFilter(unordered_map<string, unsigned int>& xMInput,
...
             loadSingleFilter(xMInput, xMSamples, *samplefilters.get(), file);
         }
     }
 
-    cout << "Info:SamplesFilter " << samplefilters->size() <<endl;
+    cout << "Info:SamplesFilter " << samplefilters->size() << endl;
 }
 
-void SamplesFilter::applyFilter(float *xArray,int xSamplesIndex, int offSet, int width)
+void SamplesFilter::applyFilter(float *xArray, int xSamplesIndex, int offset, int width)
```
Comment (rybakov) [inline src/amazon/dsstne/utils/Filters.cpp]: const int offset, const int width

File: src/amazon/dsstne/utils/Filters.cpp [hunk 3]
```diff
@@ -190,62 +176,49 @@ void SamplesFilter::loadFilter(unordered_map<string, unsigned int>& xMInput,
...
+void SamplesFilter::applyFilter(float *xArray, int xSamplesIndex)
 {
-    unordered_map<int,float> *filter= (*samplefilters)[xSamplesIndex].get();
-    updateRecords(xArray,filter);
+    unordered_map<int, float> *filter = (*samplefilters)[xSamplesIndex].get();
+    updateRecords(xArray, filter);
 }
 
-SamplesFilter::~SamplesFilter()
+FilterConfig* loadFilters(const std::string &samplesFilterFileName,
```
Comment (rybakov) [inline src/amazon/dsstne/utils/Filters.cpp]: minor: there is an inconsistency of & usage, in one case it is on the right side (const std::string &outputFileName) on other on the left (unordered_map<string, unsigned int>& xMInput)

File: src/amazon/dsstne/utils/Filters.cpp (hunk 2 above)
Comment (tristanpenman) [inline src/amazon/dsstne/utils/Filters.cpp]: As above.

File: src/amazon/dsstne/utils/Filters.cpp (hunk 3 above)
Comment (tristanpenman) [inline src/amazon/dsstne/utils/Filters.cpp]: Will leave this as-is, as discussed offline.

File: src/amazon/dsstne/utils/Filters.cpp [hunk 4]
```diff
@@ -190,62 +176,49 @@ void SamplesFilter::loadFilter(unordered_map<string, unsigned int>& xMInput,
...
+void SamplesFilter::applyFilter(float *xArray, int xSamplesIndex, int offset, int width)
 {
-    unordered_map<int,float> *filter= (*samplefilters)[xSamplesIndex].get();
-    updateRecords(xArray,filter, offSet, width);
+    unordered_map<int, float> *filter = (*samplefilters)[xSamplesIndex].get();
+    updateRecords(xArray, filter, offset, width);
 }
 
-void SamplesFilter::applyFilter(float *xArray,int xSamplesIndex)
+void SamplesFilter::applyFilter(float *xArray, int xSamplesIndex)
```
Comment (tristanpenman) [inline src/amazon/dsstne/utils/Filters.cpp]: As above.

**Pull request merged**